src/Symfony/Component/Notifier/Bridge/Mattermost/MattermostTransport.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Bridge\Mattermost;
+
+use Symfony\Component\Notifier\Exception\LogicException;
+use Symfony\Component\Notifier\Exception\TransportException;
+use Symfony\Component\Notifier\Message\ChatMessage;
+use Symfony\Component\Notifier\Message\MessageInterface;
+use Symfony\Component\Notifier\Transport\AbstractTransport;
+use Symfony\Contracts\EventDispatcher\EventDispatcherInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * @author Emanuele Panzeri <thepanz@gmail.com>
+ *
+ * @internal
+ *
+ * @experimental in 5.0
+ */
+final class MattermostTransport extends AbstractTransport
+{
+    private $token;
+    private $chatChannel;
+
+    public function __construct(string $token, string $chatChannel, HttpClientInterface $client = null, EventDispatcherInterface $dispatcher = null)
+    {
+        $this->token = $token;
+        $this->chatChannel = $chatChannel;
+        $this->client = $client;
+
+        parent::__construct($client, $dispatcher);
+    }
+
+    public function __toString(): string
+    {
+        return sprintf('mattermost://%s?channel=%s', $this->getEndpoint(), $this->chatChannel);
+    }
+
+    public function supports(MessageInterface $message): bool
+    {
+        return $message instanceof ChatMessage;
+    }
+
+    /**
+     * @see https://api.mattermost.com
+     */
+    protected function doSend(MessageInterface $message): void
+    {
+        if (!$message instanceof ChatMessage) {
+            throw new LogicException(sprintf('The "%s" transport only supports instances of "%s" (instance of "%s" given).', __CLASS__, ChatMessage::class, \get_class($message)));
+        }
+
+        $endpoint = sprintf('https://%s/api/v4/post', $this->getEndpoint());
```

Comment:
This would require users to authenticate the Mattermost notifier with a personal access token right? Wouldn't you want to use a [webhook](https://developers.mattermost.com/integrate/incoming-webhooks/) here?

Reply:
The idea is to use a Bot account, as described here: https://docs.mattermost.com/developer/bot-accounts.html

Reply:
Oh that'd be nice as well, forgot about that feature 👍.